src/builder.rs

```diff
@@ -206,14 +217,30 @@ pub enum InitError<DI, P> {
     /// This error is returned when the configuration passed to the builder is
     /// invalid. For example, when the combination of bit depth and interface
     /// kind isn't supported by the selected model.
-    InvalidConfiguration,
+    InvalidConfiguration(ConfigurationError),
+}
+
+/// Specifics of [InitError::InvalidConfiguration] if configuration was found invalid
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum ConfigurationError {
+    /// Display Model and Interface kind are mismatching
+    ModelInterfaceMismatch,
+    /// Display dimensions provided in [Builder::display_size] were invalid, e.g. width or height of 0
+    DisplayDimensions,
+    /// Display width from [Builder::display_size] + offset_x from [Builder::display_offset] > maximum width from framebuffer definition
+    XOffsetOverflow,
+    /// Display height from [Builder::display_size] + offset_y from [Builder::display_offset] > maximum height from framebuffer definition
+    YOffsetOverflow,
```

Comment:
This is probably a bit nitpicky, but I don't like the name `XOffsetOverflow, because in my mind it implies that the offset is overflowing.

How about this? I don't think we need to be as fine grained with our error reporting and instead offer a bit more explanation in the doc comment.

```suggestion
    /// Invalid display offset.
    ///
    /// The active display area, defined by
    /// [`display_size`](Builder::display_size) and
    /// [`display_offset`](Builder::display_offset), extends beyond the boundaries of
    /// the controller's framebuffer. To resolve this, reduce the offset to a
    /// maximum value of [`FRAMEBUFFER_SIZE`](Model::FRAMEBUFFER_SIZE) minus
    /// [`display_size`](Builder::display_size).
    InvalidOffset,
```